src/Symfony/Component/Messenger/Attribute/AsMessage.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Attribute;
+
+/**
+ * Attribute for configuring message routing.
+ *
+ * @author Pierre Rineau pierre.rineau@processus.org>
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
+class AsMessage
```

Comment:
I realized that the `As*` convention is used for services, and this is not a service.
Any alternative ideas? `#[Transports]`?

Reply:
I was doubtful as well, but is there any precedent in Symfony?

Ideally, in a near future, this attribute could also serve the purpose of registering name aliases (eg. `App\BlogPostCreate` -> `blog_post_create` for example) in order to allow heterogeneous (ie. non-PHP) system interactions.

I'm thinking of this in order to avoid attributes proliferation on user classes: I'd rather keep a single attribute to describe and configure a message rather than many.

Keeping this in mind, even thought messages are not and will never be services, at some point a registration mechanism would be required for this to work (otherwise it will be impossible to get incoming message PHP class from normalized alias): in this context, I think the `As*` name is right.

Any ideas, any precedence, any examples or counter-examples for this?

Once again, I'm thinking in the long term, I'm not just adding a feature, but planning a future set of coherent and coupled set of features. This one is rather simple and easy to have, but consistent naming must be thought well now, later will be too late.

Reply:
Is there someone familiar enough with the messenger component to ping for advice or opinions about this?